sqlglot/generator.py

```diff
@@ -620,8 +630,10 @@ def literal_sql(self, expression):
         if expression.is_string:
             if self._replace_backslash:
                 text = text.replace("\\", "\\\\")
-            text = text.replace(self.quote, self._escaped_quote)
-            return f"{self.quote}{text}{self.quote}"
+            text = text.replace(self.quote_start, self._escaped_quote_start)
```

Comment:
it sucks that you have to check escape_quote != end every time, maybe instead of self._escaped_quote_start and end, we store a lambda and just call that 

if ...
  self._escape_func = lambda...
else
  ...

Reply:
I'm wondering if escape_quote_start doesn't matter here actually and can collapse down. From a parsing perspective, it just can't be the escape_quote_end, right?

Reply:
oh yea i wonder if this is a bug?

Reply:
Updated to match my comment.